### PR TITLE
Feature/gitops add on configuration

### PR DESCRIPTION
**Pull request by kcoleman731**
*Issue #, if available:*

*Description of changes:*

This PR proposes changes needed to enable support for managing add-ons via GitOps with ArgoCD. From a top level module perspective, it introduces a new property `argocd_manage_add_ons`. This property indicates that add-ons should be managed via ArgoCD and that Helm charts for each enabled add-on should not be directly applied via Terraform. 

Additionally, it adds a `argocd_gitops_config` output property to the `aws-for-fluentbit`,  add-on (other add-ons will be supported in subsequent PRs). This property is used to build the values map that is the passed to the ArgoCD root Application resource. 

Lastly, customers may indicate which ArgoCD application serves as their root add-on Application via an `add_on_application` flag. ArgoCD configuration will now look like the following:

```
argocd_enable           = true
argocd_manage_add_ons   = true
argocd_applications     = {
  infra = {
    namespace             = "argocd"
    path                  = "<path>"
    repo_url              = "<repo_url>"
    target_revision       = "HEAD"
    destination           = "https://kubernetes.default.svc"
    project               = "default"
    values                = {}
    add_on_application    = true # Indicates the root add-on application. 
  }
}
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/add-ons/argocd.md [hunk 1]
```diff
@@ -49,3 +49,30 @@ argocd_applications = {
...
+    path                  = "<path>"
+    repo_url              = "<repo_url>"
+    target_revision       = "HEAD"
+    destination           = "https://kubernetes.default.svc"
+    project               = "default"
+    values                = {}
+    add_on_application    = true # Indicates the root add-on application. 
+  }
+}
+```
```
Comment (askulkarni2) [inline docs/add-ons/argocd.md:78]: Maybe just call out that alternatively if customers wish to do a CD pipeline (like GitLab) that's possible by using the default behavior.

**Review by askulkarni2 — COMMENTED** [review]
looking good so far...

File: kubernetes-addons/aws-for-fluentbit/locals.tf [hunk 2]
```diff
@@ -15,8 +16,6 @@ locals {
     namespace                                 = "kube-system"
```
Comment (vara-bonthu) [inline kubernetes-addons/aws-for-fluentbit/locals.tf:16]: We need to move Addons to its dedicated namespaces. Majority of them are suing `kube-system`. We can pick this up a seprate PR

**Review by vara-bonthu — APPROVED** [review]
LGTM

File: docs/add-ons/argocd.md (hunk 1 above)
Comment (kcoleman731) [reply docs/add-ons/argocd.md:78]: Agree - although I plan to add this content to a top level `add-ons` doc that discusses the different approaches.

File: kubernetes-addons/aws-for-fluentbit/locals.tf (hunk 2 above)
Comment (kcoleman731) [reply kubernetes-addons/aws-for-fluentbit/locals.tf:16]: agree.

**Pull request merged**